src/compiler/checker.rs

```diff
@@ -880,12 +1027,10 @@ impl<'type_checker> CheckTypeRelatedTo<'type_checker> {
         result != Ternary::False
     }
 
-    fn report_error(&self, message: &DiagnosticMessage) {
+    fn report_error(&self, message: &DiagnosticMessage, args: Option<Vec<String>>) {
```

Comment:
I represented these "error format-string arguments" (so eg the strings `true` and `number` that will be plugged into the diagnostic message `"Type '{0}' is not assignable to type '{1}'."`) (that are represented in the Typescript version as either `arg0, arg1, [...]` or variadic `...args`) as an `Option<Vec<String>>` (should probably be `Option<Vec<&str>>`)